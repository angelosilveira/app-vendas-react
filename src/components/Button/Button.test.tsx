import { screen } from '@testing-library/react';
import { renderWithTheme } from 'src/utils/tests/renderWithTheme';
import { describe, it, expect } from 'vitest';
import { BiSave } from 'react-icons/bi';

import { Button } from '.';

describe('<Button />', () => {
  it('should render the button', () => {
    const { container } = renderWithTheme(<Button>salvar</Button>);

    expect(screen.getByRole('button', { name: /salvar/i })).toHaveStyle({
      padding: '0.375rem 0.75rem',
    });
    expect(screen.getByRole('button', { name: /salvar/i })).toHaveStyle({
      width: '100%',
    });
    expect(screen.getByRole('button', { name: /salvar/i })).toHaveStyle({
      fontSize: '1.2rem',
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the button with variant', () => {
    renderWithTheme(<Button variant="success">Salvar</Button>);
    expect(screen.getByRole('button', { name: /salvar/i })).toHaveStyle({
      borderColor: '#198754',
      backgroundColor: '#198754',
    });
  });

  it('should render the button with icon', () => {
    renderWithTheme(
      <Button icon={<BiSave data-testid="icon" />}>Salvar</Button>
    );
    expect(screen.getByTestId('icon')).toBeInTheDocument();
  });
});
