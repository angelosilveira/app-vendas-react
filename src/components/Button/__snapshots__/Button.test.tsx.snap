// Vitest Snapshot v1

exports[`<Button /> > should render the button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  font-weight: 400;
  line-height: 1.5;
  color: #212529;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  border-radius: 0.25rem;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  font-size: 1.2rem;
  color: #fff;
  background-color: #0d6efd;
  border-color: #0d6efd;
}

.c0 svg {
  font-size: 1.6rem;
  margin-right: 5px;
}

<button
  class="c0"
>
  salvar
</button>
`;
